analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseTrueForAll.cs

```diff
@@ -0,0 +1,123 @@
+        //             ^^^
+
+        var nullableBad = list?.All(x => true); // Noncompliant
+
+        var good = list.TrueForAll(x => true); // Compliant
+
+        List<int> DoWork() => null;
+        Func<List<int>, bool> func = l => l.All(x => true); // Noncompliant
+
+        var methodBad = DoWork().All(x => true); // Noncompliant
+        var methodGood = DoWork().TrueForAll(x => true); // Compliant
+
+        var nullableMethodBad = DoWork()?.All(x => true); // Noncompliant
+        var nullableMethodGood = DoWork()?.TrueForAll(x => true); // Compliant
+
+        var inlineInitialization = new List<int> { 42 }.All(x => true); // Noncompliant
+
+        var imposter = new ContainsAllMethod<int>();
+        imposter.All(x => true); // Compliant
+
+        var badList = new BadList<int>();
+        badList.All(x => true); // Compliant
+
+        var goodList = new GoodList<int>();
+        goodList.All(x => true); // Noncompliant
+
+        var ternary = (true ? list : goodList).All(x => true); // Noncompliant
+        var nullCoalesce = (list ?? goodList).All(x => true); // Noncompliant
+        var ternaryNullCoalesce = (list ?? (true ? list : goodList)).All(x => true); // Noncompliant
+
+        _ = goodList.Fluent().Fluent().Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        //                                                   ^^^
+
+        All<int>(x => true); // Compliant
+        AcceptMethod<int>(goodList.All); //FP this is not an invocation, just a member access
```

Comment:
Is it `FP` or `FN`?
I think this a `FN`.

Reply:
Yes, FPs are commented like `// Noncompliant FP` to make the UT pass

Reply:
Yeap it's an FN, my mistake